pkg/kubelet/kubelet.go

```diff
@@ -2675,28 +2695,33 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 	// nodeOODCondition.Status != api.ConditionTrue or
 	// nodeOODCondition.Status != api.ConditionFalse in the conditions below depending on whether
 	// the kubelet is out of disk or not.
-	if kl.isOutOfDisk() {
-		if nodeOODCondition.Status != api.ConditionTrue {
-			nodeOODCondition.Status = api.ConditionTrue
-			nodeOODCondition.Reason = "KubeletOutOfDisk"
-			nodeOODCondition.Message = "out of disk space"
-			nodeOODCondition.LastTransitionTime = currentTime
-			kl.recordNodeStatusEvent("NodeOutOfDisk")
-		}
-	} else {
-		if nodeOODCondition.Status != api.ConditionFalse {
-			nodeOODCondition.Status = api.ConditionFalse
-			nodeOODCondition.Reason = "KubeletHasSufficientDisk"
-			nodeOODCondition.Message = "kubelet has sufficient disk space available"
-			nodeOODCondition.LastTransitionTime = currentTime
-			kl.recordNodeStatusEvent("NodeHasSufficientDisk")
-		}
+	isOutOfDisk := kl.isOutOfDisk()
+	if isOutOfDisk && nodeOODCondition.Status != api.ConditionTrue {
+		nodeOODCondition.Status = api.ConditionTrue
+		nodeOODCondition.Reason = "KubeletOutOfDisk"
+		nodeOODCondition.Message = "out of disk space"
+		nodeOODCondition.LastTransitionTime = currentTime
+		kl.recordNodeStatusEvent("NodeOutOfDisk")
+	}
+	if !isOutOfDisk && nodeOODCondition.Status != api.ConditionFalse {
+		nodeOODCondition.Status = api.ConditionFalse
+		nodeOODCondition.Reason = "KubeletHasSufficientDisk"
+		nodeOODCondition.Message = "kubelet has sufficient disk space available"
+		nodeOODCondition.LastTransitionTime = currentTime
+		kl.recordNodeStatusEvent("NodeHasSufficientDisk")
 	}
 
 	if newOODCondition {
 		node.Status.Conditions = append(node.Status.Conditions, *nodeOODCondition)
 	}
+	return
+}
 
+// Maintains Node.Spec.Unschedulable value from previous run of tryUpdateNodeStatus()
+var oldNodeUnschedulable bool
```

Comment:
this global variable is exist before, i just move here to let it closer to where it used.
:)